slicebase/bin/slice-canary

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+YUMCONF="yum.conf"
+CANARYCONF="yum.canary.conf"
+YUMREPOD="yum.slice.d"
+CANARYREPOD="yum.canary.d"
+
+if [ -s /etc/slicename ]; then
+    SLICENAME=$(cat /etc/slicename)
+else
+    echo "Error: /etc/slicename is either empty or doesn't exist."
+    exit 1
+fi
+
+if [[ $EUID -ne 0 ]] ; then
+    echo "Error: please run slice-canary as root."
+    exit 1
+fi
+
+if ! service slicectrl stop &> /dev/null; then
+    echo "Error: slicectrl failed to stop all slice services."
+    exit 1
+fi
+
+# Function to clean up temporary yum configs
+function cleanup {
+    rm -rf /tmp/$CANARYCONF /tmp/$CANARYREPOD
+}
+
+# Copy the regular production yum configurations to new "canary" versions that
+# we can use to upgrade the slice without actually modifying the production
+# versions.
+cp /etc/$YUMCONF /tmp/$CANARYCONF
+cp -a /etc/$YUMREPOD /tmp/$CANARYREPOD
+
+# Modify the "canary" yum configurations to point to the M-Lab staging yum
+# repository.
+sed -i "s/\/etc\/${YUMREPOD//\./\\.}/\/tmp\/${CANARYREPOD//\./\\.}/" \
```

Comment:
Good point.  I changed the substitution delimiter character to the pipe symbol, which eliminated the need to escape the forward slashes.  I also removed the variable substitution from the replacement pattern, since that is just a string and doesn't necessarily need to be a valid regex.  However, I've left the variable substitution on the pattern match, otherwise the string "yum.conf.d" could match "yumZconf9". I could define a whole new variable with the regex like _YUMCONFREGEX='yum\.conf\.d'_, but then every time that might change you need to update two variables instead of just one. Granted, that is probably never likely to change.  And also granted that the simple regex "yum.conf.d" is pretty much guaranteed to match every time, since we would never expect to find a string like 'yumZconf9' in the yum configuration files. But it seems like bad form.  I tried to get away with that in a code review I gave to Michael, and he rejected it on principle. :) I also added a small comment to explain the variable substitution a bit.

Do the changes I've made resolve your concern, or do you still feel it's too unreadable?